src/auth-service/models/Candidate.js

```diff
@@ -85,7 +85,7 @@ CandidateSchema.statics = {
         return {
           success: true,
           message: "no candidates exist",
-          data,
+          data: [],
```

Comment:
for cases where the category does not have any candidate, this data variable had not been declared so it was returning a runtime error.